js/ui/appDisplay.js

```diff
@@ -362,10 +362,20 @@ const EndlessApplicationView = new Lang.Class({
         return false;
     },
 
+    _reloadIcons: function() {
+        for (let icon of this._allIcons) {
+            if (icon.app) {
+                icon.app.reload_app_info();
+                icon.icon.reloadIcon();
+            }
+        }
+    },
+
     addIcons: function(isHidden) {
         // Don't do anything if we don't have more up-to-date information, since
         // re-adding icons unnecessarily can cause UX problems
         if (!this.iconsNeedRedraw()) {
+            this._reloadIcons();
```

Comment:
I have to agree with @cosimoc here: I/O in the compositor's main thread is pretty much going to cause interactivity issues on less powerful hardware platforms.